terraform/aws/aws.tf

```diff
@@ -0,0 +1,199 @@
+variable "region" { }
+variable "secret_key" { }
+variable "access_key" { }
+variable "cluster_id" { }
+variable "network_ipv4" {default = "10.0.0.0/16"}
+variable "network_subnet_ip4" {default = "10.0.0.0/16"}
+variable "control_count" {default = "3"}
+variable "worker_count" {default = "1"}
+variable "control_type" {default = "m1.small"}
+variable "source_ami" { }
+variable "worker_type" {default = "m1.small"}
+variable "public_key" { }
+
+
+provider "aws" {
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
+  region = "${var.region}"
+}
```

Comment:
We're going to approach this so that all the Terraform scripts are called as modules. Therefore, you can assume that the AWS provider will already be correctly configured and remove this. Since you'll still want to test locally, here's a config you can adapt:

```
provider "google" {
  account_file = "account.json"
  project = "asteris-mi"
  region = "us-central1"
}

module "test-cluster" {
  source = "./gce"
  datacenter = "test-cluster"
  control_type = "n1-standard-1"
  worker_type = "n1-highcpu-2"
  network_ipv4 = "10.0.0.0/16"
  long_name = "test-cluster-infrastructure"
  short_name = "test-cluster"
  region = "us-central1-a"
  control_count = 1
  worker_count = 3
}
```

I've put that in my `terraform/` directory and run it from there to use the code as a module.